hcl2template/types.build.hcp_packer_registry.go

```diff
@@ -58,7 +64,33 @@ func (p *Parser) decodeHCPRegistry(block *hcl.Block) (*HCPPackerRegistryBlock, h
 
 	par.Slug = b.Slug
 	par.Description = b.Description
-	par.Labels = b.Labels
+
+	if len(b.Labels) > 0 && len(b.BucketLabels) > 0 {
+		diags = append(diags, &hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  fmt.Sprintf("%s.labels and %[1]s.bucket_labels are mutually exclusive; please use the recommended argument %[1]s.bucket_labels", buildHCPPackerRegistryLabel),
+			Subject:  block.DefRange.Ptr(),
+		})
+		return nil, diags
+	}
+
+	if len(b.Labels) > 0 {
+		diags = append(diags, &hcl.Diagnostic{
+			Severity: hcl.DiagWarning,
+			Summary:  fmt.Sprintf("the argument %s.labels has been deprecated and will be removed in a future release; please use %[1]s.bucket_labels", buildHCPPackerRegistryLabel),
+		})
+
+		if b.BucketLabels == nil {
+			b.BucketLabels = make(map[string]string)
+		}
+
+		for k, v := range b.Labels {
+			b.BucketLabels[k] = v
+		}
```

Comment:
Could this be simplified to:
```suggestion
		b.BucketLabels = b.Labels
```
?

Reply:
Yes, good catch. Originally, I thought that I would merge Labels into BucketLabels if they were both set but that was before adding the mutually exclusion check.

Reply:
Hehe, yeah, that's what I thought happened :)